exonum-java-binding/core/src/main/java/com/exonum/binding/core/service/Schema.java

```diff
@@ -16,21 +16,22 @@
 
 package com.exonum.binding.core.service;
 
-import com.exonum.binding.common.hash.HashCode;
-import java.util.List;
+import com.exonum.binding.core.storage.indices.ProofListIndexProxy;
+import com.exonum.binding.core.storage.indices.ProofMapIndexProxy;
 
 /**
  * A schema of the collections (a.k.a. indices) of a service.
+ *
+ * <p>To verify the integrity of the database state on each node in the network,
+ * the core automatically tracks every Merkelized collection used by the user
+ * services. It aggregates state hashes of these collections into a single
+ * Merkelized meta-map. The hash of this meta-map is considered the hash of the
+ * entire blockchain state and is recorded as such in blocks and Precommit
+ * messages.
+ *
```

Comment:
The original doc had a warning on using plain indexes as a sep. paragraph:
 * > Please note that if this service does not provide any state hashes,
 *    * the framework will not be able to verify that its transactions cause the same results
 *    * on different nodes.